src/data/real/ereal.lean

```diff
@@ -783,75 +811,59 @@ by rcases lt_trichotomy 0 x with hx | rfl | hx; simp [abs_def]
-  { rcases lt_trichotomy 0 x with hx|rfl|hx,
-    { simp only [coe_mul_top_of_pos hx, hx.ne', with_top.mul_top, ne.def, abs_eq_zero_iff,
-        coe_eq_zero, not_false_iff, abs_top] },
-    { simp only [coe_zero, abs_zero, zero_mul] },
-    { simp only [coe_mul_top_of_neg hx, hx.ne, abs_bot, with_top.mul_top, ne.def, abs_eq_zero_iff,
-        coe_eq_zero, not_false_iff, abs_top] } },
-  { refl },
-  { rcases lt_trichotomy 0 y with hy|rfl|hy,
-    { simp only [top_mul_coe_of_pos hy, hy.ne', with_top.top_mul, ne.def, abs_eq_zero_iff,
-        coe_eq_zero, not_false_iff, abs_top] },
-    { simp only [coe_zero, abs_zero, mul_zero] },
-    { simp only [top_mul_coe_of_neg hy, hy.ne, abs_bot, with_top.top_mul, ne.def, abs_eq_zero_iff,
-        coe_eq_zero, not_false_iff, abs_top] } },
-  { refl }
+   -- TODO: replace with `induction using` in Lean 4, which supports multiple premises
+  with_cases
+  { apply @induction₂ (λ x y, (x * y).abs = x.abs * y.abs) };
+    propagate_tags { try { dsimp only} },
+  case [t_t, b_t, t_b, b_b] { all_goals { refl } },
+  case [t_z, b_z, z_t, z_b] { all_goals { simp only [zero_mul, mul_zero, abs_zero] } },
+  case h : x y { simp only [← coe_mul, ereal.abs, abs_mul, ennreal.of_real_mul (abs_nonneg _)], },
+  case p_b : x hx { simp only [coe_mul_bot_of_pos hx, hx.ne', abs_bot, with_top.mul_top, ne.def,
+                               abs_eq_zero_iff, coe_eq_zero, not_false_iff] },
+  case n_b : x hx { simp only [coe_mul_bot_of_neg hx, hx.ne, abs_bot, with_top.mul_top, ne.def,
+                               abs_eq_zero_iff, coe_eq_zero, not_false_iff, abs_top] },
+  case p_t : x hx { simp only [coe_mul_top_of_pos hx, hx.ne', with_top.mul_top, ne.def,
+                               abs_eq_zero_iff, coe_eq_zero, not_false_iff, abs_top] },
+  case n_t : x hx { simp only [coe_mul_top_of_neg hx, hx.ne, abs_bot, with_top.mul_top, ne.def,
+                               abs_eq_zero_iff, coe_eq_zero, not_false_iff, abs_top] },
+  case t_p : y hy { simp only [top_mul_coe_of_pos hy, hy.ne', with_top.top_mul, ne.def,
+                               abs_eq_zero_iff, coe_eq_zero, not_false_iff, abs_top] },
+  case t_n : y hy { simp only [top_mul_coe_of_neg hy, hy.ne, abs_bot, with_top.top_mul, ne.def,
+                               abs_eq_zero_iff, coe_eq_zero, not_false_iff, abs_top] },
+  case b_p : y hy { simp only [bot_mul_coe_of_pos hy, hy.ne', abs_bot, with_top.top_mul, ne.def,
+                               abs_eq_zero_iff, coe_eq_zero, not_false_iff] },
+  case b_n : y hy { simp only [bot_mul_coe_of_neg hy, hy.ne, abs_bot, with_top.top_mul, ne.def,
+                               abs_eq_zero_iff, coe_eq_zero, not_false_iff, abs_top] },
 end
 
 /-! ### Sign -/
 
-@[simp] lemma sign_top : sign (⊤ : ereal) = sign_type.pos := rfl
-@[simp] lemma sign_bot : sign (⊥ : ereal) = sign_type.neg := rfl
+@[simp] lemma sign_top : sign (⊤ : ereal) = 1 := rfl
+@[simp] lemma sign_bot : sign (⊥ : ereal) = -1 := rfl
```

Comment:
`pos` and `neg` are not intended to be used outside of the `sign` file, so this swaps them to the simp-normal form.